### ISSUE TITLE
`Promise To` does not work with `*args`

### ISSUE THREAD

**Issue by mbondarew** [MarketSquare/robotframework-browser#3200]
**Describe the bug**
After updating to the latest version of the Browser library, I started getting errors when using both the `Promise To` keyword and keywords from my extension that can take multiple arguments. Thank you!

**To Reproduce**
```js
// extension.js

exports.__esModule = true

// by the documentation this format with 'args' is supported, isn't it?
exports.waitForSomething = async function (page, args, logger) {
   logger(args.join())
}
```
```robot
*** Settings ***
Library     	Browser  jsextension=${CURDIR}/extension.js

*** Test Cases ***
Promise To and Keyword that takes multiple Arguments
	# without Promise To next two lines work
	Wait For Something	one
	Wait For Something	one		two
	# with Promise To and one argument it also works
	Promise To    Wait For Something	one
	# but with Promise To and two arguments --> IndexError: list index out of range, 
	Promise To    Wait For Something	one		two
```

Result:
```log
Promise To and Keyword that takes multiple Arguments                  | FAIL |
IndexError: list index out of range
```

Debug:
```log
20231031 18:21:42.581 - INFO - +-- START KEYWORD: Browser.Promise To [ Wait For Something | one | two ]
20231031 18:21:42.582 - DEBUG - *args ('one', 'two')
20231031 18:21:42.582 - FAIL - IndexError: list index out of range
20231031 18:21:42.589 - DEBUG - Traceback (most recent call last):
  File "/home/user/.local/lib/python3.10/site-packages/Browser/browser.py", line 1085, in run_keyword
    return DynamicCore.run_keyword(self, name, args, kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/robotlibcore.py", line 126, in run_keyword
    return self.keywords[name](*args, **(kwargs or {}))
  File "/home/user/.local/lib/python3.10/site-packages/Browser/keywords/promises.py", line 63, in promise_to
    positional, named = self.resolve_arguments(known_keyword, *args)
  File "/home/user/.local/lib/python3.10/site-packages/Browser/keywords/promises.py", line 101, in resolve_arguments
    positional.append(self.convert_keyword_arg(kw, arg_names[index], arg))
IndexError: list index out of range
20231031 18:21:42.590 - INFO - +-- END KEYWORD: Browser.Promise To (8)
```

**Expected behavior**
Execution without errors

**Desktop:**
 - OS: Linux Kubuntu 22.04

**Comment by aaltat** [comment]
@allcontributors please add @mbondarew for bugs

**Comment by Snooz82** [comment]
I changed the title, because this is a general problem when using *varargs in any keyword with `Promise To`

**Comment by Snooz82** [comment]
fixed by #3212
